backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/externalsystem/MetasfreshToExternalSystemRabbitMQ_StepDef.java

```diff
@@ -204,4 +224,46 @@ public void handleDelivery(final String consumerTag, final Envelope envelope, fi
 				})
 				.collect(ImmutableList.toImmutableList());
 	}
+
+	private void validateExportPPOrderWithPLUFile(
+			@NonNull final String ppOrderIdentifier,
+			@NonNull final List<JsonExternalSystemRequest> requests,
+			@NonNull final I_ExternalSystem_Config externalSystemConfig,
+			@NonNull final Map<String, String> tableRow) throws JsonProcessingException
+	{
+		final I_PP_Order ppOrder = ppOrderTable.get(ppOrderIdentifier);
+
+		final JsonExternalSystemRequest jsonExternalSystemRequest = requests.stream()
+				.filter(request -> request.getExternalSystemConfigId().getValue() == externalSystemConfig.getExternalSystem_Config_ID())
+				.filter(request -> request.getParameters().get(PARAM_PP_ORDER_ID) != null)
+				.filter(request -> Integer.parseInt(request.getParameters().get(PARAM_PP_ORDER_ID)) == ppOrder.getPP_Order_ID())
+				.findFirst()
+				.orElse(null);
+
+		assertThat(jsonExternalSystemRequest).isNotNull();
```

Comment:
i had this pattern on friday, where the assertion failed and it gave me a ton of headache to figure out the problem (in the end, one of those filters used BigDecimal.equals instead of compareTo).

Please make sure that instead of just failing you
- display the requests you looked at
- display why they were ruled out

Reply:
logs to display why the request that was filtered out added